ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -134,6 +142,16 @@ class CKTypedComponentAction : public CKTypedComponentActionBase {
   }
 
   /**
+   Allows demoting an action to a simpler action while supplying defaults for the values that won't be passed in.
+   */
+  template<typename... U, bool separator = false, class = typename std::enable_if<!detail::any_are_reference<U...>::value>::type>
```

Comment:
Good call - using the static_assert simplifies things and provides a better error message. Thanks for pointing this out!